src/actions/__tests__/read-rule.test.js

```diff
@@ -74,4 +74,45 @@ describe('readRule()', () => {
       runReadRule(undefined, docs.replace('\n', '\r\n')).newDocs
     ).toMatchSnapshot()
   })
+
+  it('mixed windows and unix newlines', () => {
+    const docs = `# ${description} (${name})\nSome random test:\r\n\r\n`
+
+    expect(runReadRule(undefined, docs).newDocs).toBe(docs)
+  })
+
+  it('whitespaces before heading', () => {
+    const docs = `\n\n\n\n\r\n# ${description} (${name})\n`
+    const outputDocs = `# ${description} (${name})\n`
+
+    expect(runReadRule(undefined, docs).newDocs).toBe(outputDocs)
+  })
+
+  it('non string text before heading', () => {
+    const docs = `\nTest\n`
+    const outputDocs = `# ${description} (${name})\n\nTest\n`
+
+    expect(runReadRule(undefined, docs).newDocs).toBe(outputDocs)
+  })
+
+  it('non string text before heading without new line', () => {
+    const docs = `Test\n`
+    const outputDocs = `# ${description} (${name})\nTest\n`
+
+    expect(runReadRule(undefined, docs).newDocs).toBe(outputDocs)
+  })
+
+  it('single line heading', () => {
+    const docs = `# Test`
+    const outputDocs = `# ${description} (${name})`
+
+    expect(runReadRule(undefined, docs).newDocs).toBe(outputDocs)
+  })
+
+  it('empty file', () => {
```

Comment:
Can you rephrase these so they form complete sentences?

```suggestion
  it('handles an empty file properly', () => {
```

(If you need assistance, please don’t hesitate to ask for help)

Reply:
Looks like these changes didn’t get pushed.

Reply:
i see them in github https://github.com/j-f1/eslint-docs/pull/37/files#diff-1ef53d81915ab858a1a34aa1c5735e25R113

Reply:
Ah, I see. Can you update the other test names too?

Reply:
i did update it a little, can you check and give me your remarks?

i hate naming stuff